tests/testlib/s2n_connection_test_utils.c

```diff
@@ -37,7 +37,7 @@ int s2n_fd_set_non_blocking(int fd)
 static int buffer_read(void *io_context, uint8_t *buf, uint32_t len)
 {
     struct s2n_stuffer *in_buf;
-    int n_read, n_avail;
+    uint32_t n_read, n_avail;
```

Comment:
Do we have concerns about masking internal errors here?
It should never be the case that read_cursor > write_cursor, but by declaring these variables uint32_t we invoke undefined behavior in that error case. 
```
#define s2n_stuffer_data_available(s)  ((s)->write_cursor - (s)->read_cursor)
```

Reply:
https://github.com/aws/s2n-tls/pull/3826#discussion_r1107741170